daeun/lv1/예산.js

```diff
@@ -0,0 +1,22 @@
+function solution(d, budget) {
+    let answer = 0;
+    const sortedD = [...d].sort((a, b) => a - b)
+
+    for (let i in sortedD) {        
+        budget -= sortedD[i]
+        if (0 > budget) break;
+        answer ++;
+    }
+    return answer;
+}
+
+// reduce를 활용한 방법.
+function solution2(d, budget) {
+    const sortedD = d.sort((a, b) => a - b)
+    return sortedD.reduce((acc, cur) => {
+        budget -= cur;
+         // (budget >= 0)은 budget -= cur의 횟수만큼 그 결과를 true 혹은 false로 반환한다.
+         // 예시 1의 경우 true(1)가 세 번, false(0)가 세 번 출력되므로 acc + 3이 결과로 return 된다.
+        return acc + (budget >= 0);
+    }, 0)
+}
```

Comment:
```suggestion
// reduce를 활용한 방법.
function solution2(d, budget) {
    const sortedD = d.sort((a, b) => a - b)
    return sortedD.reduce((acc, cur) => {
        budget -= cur;
         // (budget >= 0)은 budget -= cur의 횟수만큼 그 결과를 true 혹은 false로 반환한다.
         // 예시 1의 경우 true(1)가 세 번, false(0)가 세 번 출력되므로 acc + 3이 결과로 return 된다.
        return acc + (budget >= 0);
    }, 0)
}
```
오!! (budget >= 0)이 true, false로 반환하는 걸 활용해서 사용할 수도 있군요!!
대운님 덕분에 또 새로운 방법을 알아갑니다~!! 😃👍

Reply:
신박한 방법이긴 하지만 저처럼 초보자는 한 눈에 알아보기 쉽지 않았습니다 ㅎㅎ.. 코멘트 감사합니다~!